src/fileio/datafile.cxx

```diff
@@ -98,6 +105,8 @@ Datafile& Datafile::operator=(Datafile &&rhs) {
   enabled      = rhs.enabled;
   init_missing = rhs.init_missing;
   shiftOutput  = rhs.shiftOutput;
+  writeFlushCounter = rhs.writeFlushCounter;
```

Comment:
Yes, I think that should make sure the file always gets opened.

I am not sure whether we should make sure the file gets closed in the destructor. It seems to me this is currently not the case for files that are not openclose - did I miss something?